### DISCUSSION TITLE
The `PostgresVectorStore` should have `MetadataFilter` for empty values

### DISCUSSION THREAD

**Discussion by kasteph** [run-llama/llama_index#13147] [Ideas]
From the [docs](https://docs.llamaindex.ai/en/stable/examples/vector_stores/postgres/#apply-metadata-filters), we can do:

```
...

filters = MetadataFilters(
    filters=[
        MetadataFilter(key="author", value="mats@timescale.com"),
        MetadataFilter(key="author", value="sven@timescale.com"),
    ],
    condition="or",
)

...
```

But we should also be able to do something like:

```
...

filters = MetadataFilters(
    filters=[
        MetadataFilter(key="author", value=None),
        MetadataFilter(key="author", value="sven@timescale.com"),
    ],
    condition="or",
)

...
```

where the value can be `None` since there are many cases in the real world where a filter can be `None`. For example:

1. Some documents might have no author attributed
2. Some documents might have some kind of other metadata such as `university` and others do not

and so on.

**Comment by slyapustin** [comment]
Bump this. Run into that limitation as sometimes VectorIndexRetriever generate IS_EMPTY filters with `None` as a value, which rases exception: 
> TypeError: float() argument must be a string or a real number, not 'NoneType'

#17250

**Comment by ericelizes** [comment]
Is there a solution to this?

**Comment by kasteph** [comment]
I'd be happy to write a PR if the maintainers agree to this discussion
